### ISSUE TITLE
Why does the output have one more channel than the num of landmarks?

### ISSUE THREAD

**Issue by ty2806** [protossw512/AdaptiveWingLoss#13]
In your code of the model you have

> self.add_module('l' + str(hg_module), nn.Conv2d(256,
>                                                             num_landmarks+1, kernel_size=1, stride=1, padding=0))

and in evaler.py the last channel isn't used

> pred_heatmap = outputs[-1][:, :-1, :, :][i].detach().cpu()

So I guess it's used somewhere in the loss?

**Comment by ty2806** [comment]
I just read the paper again and now my understanding is the additional channel of the heatmap output is a heatmap of face boundery and is used as a part of the adaptive wing loss.
And your model also has a boundery_channels output which I think is the x-, y- bounderies generated from coordinate channels. now my question is what loss is used for this boundery output?
